filechooser-portal/Main.vala

```diff
@@ -232,9 +220,20 @@ public class Files.FileChooserPortal : Object {
                     if (dialog.filter != null) {
                         _results["current_filter"] = dialog.filter.to_gvariant ();
                     }
-
-                    _response = 0;
-                    break;
+                    var chosen_uri = dialog.get_uri ();
+                    if (chosen_uri != supplied_uri) {
+                        check_overwrite_uri.begin (chosen_uri, (obj, res) => {
+                            if (check_overwrite_uri.end (res)) {
+                                _response = 0;
+                                save_file.callback ();
+                            }
+                        });
+
+                        return; // Continue showing dialog until check completes
+                    } else { // No need to check full uri supplied by calling app
```

Comment:
Are you sure this is safe? I can think of cases where an app-supplied URI could conflict. Say that an app provides a "save as" function and passes the existing file as a sample URI for the dialog. In that case, the app-supplied URI would create a file conflict that we would want to check for. For example, I believe that MATLAB does this behavior of passing the current file to save as, at least on Mac (I assume it's the same on other platforms).

If this is something expected by the portal that I'm missing, then you can close this question.

Reply:
I think that if the app supplies a URI then it can be assumed that the URI is in fact usable without further check. If not, then that is a problem with the app.  @Marukesu  may wish to comment as he suggested omitting the check.